src/shims/intrinsics.rs

```diff
@@ -404,32 +404,32 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
             // Other
             "exact_div" => {
-                let &[num, denom] = check_arg_count(args)?;
-                this.exact_div(this.read_immediate(num)?, this.read_immediate(denom)?, dest)?;
+                let &[ref num, ref denom] = check_arg_count(args)?;
+                this.exact_div(&this.read_immediate(num)?, &this.read_immediate(denom)?, dest)?;
             }
 
             "try" => return this.handle_try(args, dest, ret),
 
             name => throw_unsup_format!("unimplemented intrinsic: {}", name),
         }
 
-        trace!("{:?}", this.dump_place(*dest));
+        trace!("{:?}", this.dump_place(**dest));
         this.go_to_block(ret);
         Ok(())
     }
 
     fn atomic_load(
-        &mut self, args: &[OpTy<'tcx, Tag>], dest: PlaceTy<'tcx, Tag>,
+        &mut self, args: &[OpTy<'tcx, Tag>], dest: &PlaceTy<'tcx, Tag>,
         atomic: AtomicReadOp
     ) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
 
 
-        let &[place] = check_arg_count(args)?;
+        let &[ref place] = check_arg_count(args)?;
         let place = this.deref_operand(place)?;
 
         // make sure it fits into a scalar; otherwise it cannot be atomic
-        let val = this.read_scalar_atomic(place, atomic)?;
+        let val = this.read_scalar_atomic(&place, atomic)?;
```

Comment:
Why is it `deref_operand(place)` a few lines above but `read_scalar_atomic(&place` here?

Reply:
The first `ref place` binding is being shadowed by `let place = this.deref_operand(place)?;`.

Reply:
Oh, good point, thanks.